gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/FeatureConverter.java

```diff
@@ -56,6 +66,30 @@ static Feature toFeature(Project project, DocumentSnapshot doc) throws DataStore
     throw new DataStoreException("No geometry in remote feature " + doc.getId());
   }
 
+  private static PolygonFeature toFeatureFromGeometry(
+      Project project, DocumentSnapshot doc, FeatureDocument f) {
+    Map<String, Object> geometry = f.getGeometry();
+    Object type = geometry.get(GEOMETRY_TYPE);
+    if (!POLYGON_TYPE.equals(type)) {
+      throw new DataStoreException("Unknown geometry type in feature " + doc.getId() + ": " + type);
+    }
+    Object coordinates = geometry.get(GEOMETRY_COORDINATES);
+    if (coordinates == null || !coordinates.getClass().isArray()) {
+      throw new DataStoreException(
+          "Invalid coordinates in feature " + doc.getId() + ": " + coordinates);
+    }
+    ImmutableList.Builder<Point> vertices = ImmutableList.builder();
+    for (Object point : (Object[]) coordinates) {
+      if(point instanceof GeoPoint){
```

Comment:
Please use `google-java-format` to format all files as discussed.

Reply:
using checkcode.